File: 0788-rotated-digits/0788-rotated-digits.py
class Solution:
    def isGoodNumber(self, num) -> bool :
        change_possible = False

        while(num > 0) :
            digit = (num % 10)

            if(digit == 3 or digit == 4 or digit == 7) :
                return False
            
            if(digit == 2 or digit == 5 or digit == 6 or digit == 9) :
                change_possible = True
            
            num //= 10
        
        return change_possible

    def rotatedDigits(self, n: int) -> int:
        i, countGoodNumber = 1, 0

        while(i <= n) :
            if(self.isGoodNumber(i)) :
                countGoodNumber += 1
            i += 1

        return countGoodNumber

File: 0788-rotated-digits/0788-rotated-digits.java
class Solution {
    public boolean isGoodNumber(int num){
        boolean changePossible = false;

        while(num > 0){
            int digit = (num % 10);

            if(digit == 3 || digit == 4 || digit == 7) return false;
            if(digit == 2 || digit == 5 || digit == 6 || digit == 9) changePossible = true;

            num /= 10;
        }

        return changePossible;
    }

    public int rotatedDigits(int n) {
        /*
            Time complexity = O(nlog_10(n)), 
            Space complexity = O(1)
        */

        int i = 1, countGoodNumber = 0;
        
        while(i <= n){
            if(isGoodNumber(i)) countGoodNumber++;
            i++;
        }

        return countGoodNumber;
    }
}